general/device.hpp

```diff
@@ -262,10 +261,26 @@ class Device
    /** @deprecated Use GetDeviceMemoryClass() instead. */
    static inline MemoryClass GetMemoryClass() { return Get().device_mem_class; }
 
+   /** @brief Get the current Device Temporary MemoryType. This is the MemoryType used by
+       MFEM classes when allocating temporary memory to be used with device kernels.
+   */
+   static inline MemoryType GetDeviceTempMemoryType() { return Get().device_temp_mem_type; }
+
+   /** @brief Get the current Device Temporary MemoryClass. This is the MemoryClass used
+       by MFEM device kernels when they need to access temporary Memory objects. */
+   static inline MemoryClass GetDeviceTempMemoryClass() { return Get().device_temp_mem_class; }
+
    static void SetGPUAwareMPI(const bool force = true)
    { Get().mpi_gpu_aware = force; }
 
    static bool GetGPUAwareMPI() { return Get().mpi_gpu_aware; }
+
+#ifdef MFEM_USE_UMPIRE
+   static bool UseHostUmpire() { return Get().use_host_umpire; }
+   static void UseHostUmpire(bool use) { Get().use_host_umpire = use; }
+   static bool UseDeviceUmpire() { return Get().use_device_umpire; }
+   static void UseDeviceUmpire(bool use) { Get().use_device_umpire = use; }
```

Comment:
I personally prefer these to have `Get`/`Set` prefixes similar to other methods in this class.